src/main.rs

```diff
@@ -98,9 +98,30 @@ async fn try_main() -> eyre::Result<bool> {
     // Set up the HTTP client
     let connect_timeout = Duration::from_secs(10);
     let timeout = Duration::from_secs(30);
-    let client = Client::builder()
+    let mut client_builder = Client::builder()
         .connect_timeout(connect_timeout)
-        .timeout(timeout)
+        .timeout(timeout);
+
+    // Add proxy if provided
+    let proxy_opt = config
+        .rsspls
+        .proxy
+        .or(env::var("http_proxy").ok())
+        .or(env::var("HTTPS_PROXY").ok());
+    if let Some(proxy) = proxy_opt {
+        let proxy_protocol = proxy.split(":").next();
```

Comment:
I believe the proxy needs to be a valid URL. I'd suggest parsing it as a `Url` and then matching on [scheme](https://docs.rs/reqwest/0.11.24/reqwest/struct.Url.html#method.scheme). I think `reqwest::Proxy::*` will accept the parsed Url too.

Reply:
Sounds good